gpio: en7523: enable for EcoNet MIPS platform

The Airoha EN7523 GPIO controller is also found on EcoNet EN7528
MIPS SoC. Allow building the driver when ECONET is selected.

Signed-off-by: Ahmed Naseef <naseefkm@gmail.com>
--- a/drivers/gpio/Kconfig
+++ b/drivers/gpio/Kconfig
@@ -271,7 +271,7 @@ config GPIO_EM
 
 config GPIO_EN7523
 	tristate "Airoha GPIO support"
-	depends on ARCH_AIROHA
+	depends on ARCH_AIROHA || ECONET
 	default ARCH_AIROHA
 	select GPIO_GENERIC
 	select GPIOLIB_IRQCHIP
